### PR TITLE
Clear order promotions in Omnes event

### PR DESCRIPTION
**Pull request by mamhoff**
## Summary

This moves clearing the connected order promotions upon emptying an order into an `Omnes::Subscriber`. This saves us from having to override `Spree::Order#empty!` in the `solidus_legacy_promotions` gem. 

## Checklist

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

### CODE REVIEW

**Review by adammathys — APPROVED** [review]

**Review by tvdeyen — APPROVED** [review]
Nice pattern, that we should use more often

File: core/app/models/spree/order.rb [hunk 1]
```diff
@@ -450,7 +450,7 @@ def empty!
       line_items.destroy_all
       adjustments.destroy_all
       shipments.destroy_all
-      order_promotions.destroy_all
+      Spree::Bus.publish :order_emptied, order: self
```
Comment (kennyadsl) [inline core/app/models/spree/order.rb:453]: I can't recall exactly how it works.. is this always sync? Worried that the subscriber could be executed after the order is already recalculated.

Comment (mamhoff) [reply core/app/models/spree/order.rb:453]: There's an ActiveJob and Sidekiq adapter for subscribers. If those are used in the subscriber, sync can't be guaranteed. However, the particular subscriber I'm adding here uses the `method` adapter, which is synchronous.

Comment (kennyadsl) [reply core/app/models/spree/order.rb:453]: Ok, nice! I didn't remember that the adapter configuration was set at the subscriber level.

**Review by kennyadsl — APPROVED** [review]
That's the way to go! Thanks again Martin!

**Pull request merged**